plugins/catalog-import/README.md

```diff
@@ -1,21 +1,86 @@
-# Catalog import plugin
+# Catalog Import
 
-Welcome to the catalog-import plugin!
+The Catalog Import Plugin provides a wizard to onboard projects with existing `catalog-info.yaml` files.
+It also assists by creating pull requests in repositories where no `catalog-info.yaml` exists.
 
-This plugin allows you to bootstrap a component-config YAML file for your repository and open a pull request to add it.
+![Catalog Import Plugin](./docs/catalog-import-screenshot.png)
 
-When installed it is accessible on [localhost:3000/catalog-import](localhost:3000/catalog-import).
+Current features:
 
-<img src="./src/assets/catalog-import-screenshot.png" />
+- Import `catalog-info.yaml` files from a URL in a repository of one of the supported Git integrations (example `https://github.com/backstage/backstage/catalog-info.yaml`).
+- _[GitHub only]_ Search for all `catalog-info.yaml` files in a Git repository (example: `https://github.com/backstage/backstage`).
+- _[GitHub only]_ Analyze a repository, generate a Component entity, and create a Pull Request to onboard the repository.
 
-## Running
+Some features are not yet available for all supported Git providers.
 
-Just run the backstage.
+## Getting Started
 
+1. Install the Catalog Import Plugin:
+
+```bash
+# packages/app
+
+yarn add @backstage/plugin-catalog-import
+```
+
+2. Add the plugin to the app:
+
+```ts
+// packages/app/src/plugins.ts
+
+export { catalogImportPlugin } from '@backstage/plugin-catalog-import';
+```
+
+3. Register the `CatalogImportPage` at the `/catalog-import` path:
+
+```tsx
+// packages/app/src/App.tsx
+
+import { CatalogImportPage } from '@backstage/plugin-catalog-import';
+
+<Route path="/catalog-import" element={<CatalogImportPage />} />;
 ```
-yarn start && yarn --cwd packages/backend start
+
+## Customizations
+
+### Disable the creation of Pull Requests
+
+The pull request feature can be disabled by options that are passed to the `CatalogImportPage`:
+
+```tsx
+// packages/app/src/App.tsx
+
+<Route
+  path="/catalog-import"
+  element={<CatalogImportPage options={{ pullRequest: { disable: true } }} />}
+/>
+```
+
+### Customize the title and body of the Pull Request
+
+The pull request form is filled with a default title and body.
+This can be configured by options that are passed to the `CatalogImportPage`:
+
+```tsx
+// packages/app/src/App.tsx
+
+<Route
+  path="/catalog-import"
+  element={
+    <CatalogImportPage
+      options={{
+        pullRequest: {
```

Comment:
@dhenneke Realized that this perhaps adds an unnecessary layer, as I think `props` could already be considered to be options. What do you think about moving `pullRequest` up to be a prop?

Wouldn't be a breaking change as long as it goes in before the release tomorrow :grin:

@backstage/maintainers thoughts on this general pattern as well? i.e. flatter vs deeper prop structures

Reply:
Sure. I prepare a PR later